librz/util/bitvector.c

```diff
@@ -1507,3 +1507,162 @@ RZ_API bool rz_bv_is_all_one(RZ_NONNULL const RzBitVector *x) {
 	}
 	return true;
 }
+
+/**
+ * get predecessor of bv (dec 1) in 2^n modulo
+ * \param bv
+ * \return predecessor of bv
+ */
+RZ_API RZ_OWN RzBitVector *rz_bv_pred(RZ_NONNULL RzBitVector *bv) {
+	rz_return_val_if_fail(bv, NULL);
+	ut32 len = bv->len;
+	if (len <= 64) {
+		ut64 val = rz_bv_to_ut64(bv);
+		val -= 1;
```

Comment:
So what should pred 0 be?

Reply:
I suppose it would be `0xfff...` (for `len` bits), given that we are doing `pred` modulo `2^n`.

Reply:
> So what should pred 0 be?

pred 0 gets 0xffff... according to bap description and its implementation

Reply:
add a test